PocketKit/Sources/Sync/Operations/SearchService.swift

```diff
@@ -46,6 +51,14 @@ public class PocketSearchService: SearchService {
         } catch {
             // TODO: How to handle errors
             Crashlogger.capture(error: error)
+            if case URLSessionClient.URLSessionClientError.networkError(_, _, let underlying) = error {
+                switch (underlying as NSError).code {
+                case -1009:
+                    throw SearchServiceError.noInternet
```

Comment:
good question! Is it common to switch on error codes that are set as a constant? I agree and I also thought we should express what that code means, so my attempt was to create that `SearchServiceError` to show that we thrown a `noInternet` error. Also I wanted a clean way to handle the no internet connection error that was thrown by Apollo. Do you have an alternative of how to handle it compared to what I have now?

`guard case SearchServiceError.noInternet = error else { return }`

Reply:
FYI this is the error that comes back from Apollo
![image](https://user-images.githubusercontent.com/6743397/217335385-dcf7fd50-8e12-4afa-977d-58d4873b5562.png)

Reply:
Thanks for clarifying @cyndichin ! I don't think this should block the PR. I'd say the main reason to avoid the mapping would be that we don't have control over the codes generated by Apollo, but on the other end, we are displaying a message to the user and we want to be clear about what's happening. 
We could try and see what other codes behind that Apollo `networkError` mean, but, if we need to explicitly discern the no internet, then whatever logic we use, we'd need to capture that `-1009` somehow.
If you'd agree, we can merge this as it is (I am going to approve this), maybe just adding a `static let noConnectionErrorCode = -1009`, add a comment to it explaining that it tracks the `no internet` situation, and use that constant in the `case` inside `search(...)`.
Afterwards, as a point of improvement, we can dig deeper into the Apollo error codes and see if, for example, the `networkError` suffices to communicate a connection error to users. 
What do you think?

A couple of other things:
1. `Crashlogger.capture(error: error)` seems redundant, as it happens inside `search(...)` and when we catch the error thrown by it. Maybe we only want to do it outside, so that we can capture the mapped `noInternet` error?
2. Not related to this PR, but I see that the `capuure(error: Error)` method prints an `error.localizedDescription`:
    - it's probably better to just print the error, usually gives more info than the localized description
    - do we have a console logger instead of `print(...)`

Reply:
thanks for the details!~ I updated to use the constant and added a comment. I left the logic `guard case SearchServiceError.noInternet = error else { return }` so the local error is still being used. I agree that we can dig deeper into Apollo error codes and see if it suffices. 

1. I think we want to capture errors in all cases so that it gets sent to Sentry, but it is redundant in `OnlineSearch`, so I removed it from there.
2. We do not have a logger from what I know aside from sending information to Sentry, but we definitely should. I'm not sure if that was done to make the console less verbose, but we should probably be using a logger instead of print.